src/Features/VisualBasic/Portable/ConvertAutoPropertyToFullProperty/VisualBasicConvertAutoPropertyToFullProperty.vb

```diff
@@ -0,0 +1,104 @@
+
+        Friend Overrides Function GetProperty(token As SyntaxToken) As SyntaxNode
+            Dim containingProperty = token.Parent.FirstAncestorOrSelf(Of PropertyStatementSyntax)
+            If (containingProperty Is Nothing) Then
+                Return Nothing
+            End If
+
+            Dim start = If(containingProperty.AttributeLists.Count > 0,
+                containingProperty.AttributeLists.Last().GetLastToken().GetNextToken().SpanStart,
+                containingProperty.SpanStart)
+
+            ' Offer this refactoring anywhere in the signature of the property.
+            Dim position = token.SpanStart
+            If (position < start) Then
+                Return Nothing
+            End If
+
+            If containingProperty.HasReturnType() AndAlso
+                position > containingProperty.GetReturnType().Span.End Then
+                Return Nothing
+            End If
+
+            Return containingProperty
+        End Function
+
+        Friend Overrides Function GetNewAccessorsAsync(
+                document As Document,
+                propertyNode As SyntaxNode,
+                fieldName As String,
+                generator As SyntaxGenerator,
+                cancellationToken As CancellationToken) As Task(Of (newGetAccessor As SyntaxNode, newSetAccessor As SyntaxNode))
+
+            Dim returnStatement = New SyntaxList(Of StatementSyntax)(DirectCast(generator.ReturnStatement(generator.IdentifierName(fieldName)), StatementSyntax))
+            Dim getAccessor As SyntaxNode = SyntaxFactory.GetAccessorBlock(
+                    SyntaxFactory.GetAccessorStatement(),
+                    returnStatement)
+
+            Dim propertySyntax = DirectCast(propertyNode, PropertyStatementSyntax)
+
+            Dim setAccessor As SyntaxNode
+            If IsReadOnly(propertySyntax) Then
+                setAccessor = Nothing
+            Else
+                Dim setStatement = New SyntaxList(Of StatementSyntax)(DirectCast(generator.ExpressionStatement(generator.AssignmentStatement(
+                        generator.IdentifierName(fieldName),
+                        generator.IdentifierName("Value"))), StatementSyntax))
+                setAccessor = SyntaxFactory.SetAccessorBlock(
+                        SyntaxFactory.SetAccessorStatement(),
+                        setStatement)
+            End If
+
+            Return Task.FromResult((getAccessor, setAccessor))
+        End Function
+
+        Private Function IsReadOnly(propertySyntax As PropertyStatementSyntax) As Boolean
+
+            Dim modifiers = propertySyntax.GetModifiers()
+            For Each modifier In modifiers
+                If modifier.IsKind(SyntaxKind.ReadOnlyKeyword) Then
+                    Return True
+                End If
+            Next
+
+            Return False
+        End Function
+
+        Friend Overrides Function GetUniqueName(fieldName As String, propertySymbol As IPropertySymbol) As String
```

Comment:
For VB do we even want to generate a unique name?  shouldn't we always just use the actual VB name that was created.  There may be existing code that refers to that name, so we don't want to break that code.

Reply:
Changed the method to return the name of the implicit backing field.